sw/device/tests/dif/dif_csrng_smoketest.c

```diff
@@ -13,87 +13,145 @@
 
 const test_config_t kTestConfig;
 
+enum {
+  kExpectedOutputLen = 16,
+};
+
 /**
- * FIPS CAVP test vector for CTR DRBG.
+ * Compare `num_words` of `got` against `expected` words.
+ *
+ * Prints differences between `got` and `expected` before generating a test
+ * assertion if a mismatch is found.
  *
- * CAVP test parameters.
- * Cipher: AES-256
- * Derivation Function: No
- * Prediction Resistance: False
+ * @param got Input buffer.
+ * @param expected Buffer containing expected values.
+ * @param num_words Number of words to compare.
  */
-const dif_csrng_seed_material_t kEntropyInput = {
-    .seed_material =
-        {
-            0xe4bc23c5,
-            0x089a19d8,
-            0x6f4119cb,
-            0x3fa08c0a,
-            0x4991e0a1,
-            0xdef17e10,
-            0x1e4c14d9,
-            0xc323460a,
-            0x7c2fb58e,
-            0x0b086c6c,
-            0x57b55f56,
-            0xcae25bad,
-        },
-    .seed_material_len = 12,
-};
+static void check_buffer(const uint32_t *got, const uint32_t *expected,
+                         size_t num_words) {
+  if (memcmp(got, expected, num_words * sizeof(uint32_t)) == 0) {
+    return;
+  }
+
+  for (uint32_t i = 0; i < num_words; ++i) {
+    LOG_INFO("[%d] got = 0x%x; expected = 0x%x", i, got[i], expected[i]);
+  }
+  CHECK(false, "Detected buffer mismatch.");
```

Comment:
Can we factor this whole function out into some testing header (pref. as a macro to capture line numbers)? People keep seeming to have to write it.

Reply:
lowrisc/opentitan#7481